optimum/exporters/executorch/utils.py

```diff
@@ -126,3 +130,67 @@ def verify_eos_tokens_in_pretrained_tokenizer(model_eos_ids: List[int], tokenize
     is_valid = any(model_id in candidate_eos_ids for model_id in model_eos_ids)
 
     return is_valid
+
+
+def process_conversation_inputs(
+    processor: ProcessorMixin,
+    tokenizer: PreTrainedTokenizer,
+    input_conversation: List[Dict],
+):
+    """
+    Process input conversation for multimodal models.
+
+    This function handles the preprocessing of conversation inputs, with special handling for
+    GraniteSpeechProcessor which requires extracting and processing audio content from conversations
+    prior to feeding into the processor.
+
+    Args:
+        processor: The processor to use for input processing
+        tokenizer: The tokenizer to use for text processing
+        input_conversation: List of conversation messages, may contain audio content
+
+    Returns:
+        Processed inputs ready for model consumption
+    """
+    if isinstance(processor, transformers.models.granite_speech.processing_granite_speech.GraniteSpeechProcessor):
+        import requests
+        import torchaudio
+
+        conversation = copy.deepcopy(input_conversation)
+        audio_path = None
+
+        # Extract audio content and remove from conversation
+        for i, item in enumerate(conversation):
+            if "type" in item and item["type"] == "audio":
+                audio_path = item["content"]
+                # Remove the audio content from the input conversation since it
+                # is handled outside for Granite.
+                conversation.pop(i)
+                break
+
+        if audio_path is None:
+            raise ValueError("No audio content found in conversation for GraniteSpeechProcessor")
+
+        # Download and process audio
+        resp = requests.get(audio_path)
+        resp.raise_for_status()
+        buf = io.BytesIO(resp.content)
```

Comment:
Can't we have an option to just use a local audio_file? Always connecting to the internet is not great

Reply:
```
if audio_path.startswith(('http://', 'https://'))
  try 
     requests.get(audio_path)
   catch
   ....
   wav, sampling_rate = torchaudio.load(audio_path, normalize=True)
elif os.path.exists(audio_path):
   wav, sampling_rate = torchaudio.load(audio_path, normalize=True)
else:

Reply:
I think we can do this in another pr, also I'd prefer to stay consistent with export inputs